### DISCUSSION TITLE
[Zoe] List of contracts to write

### DISCUSSION THREAD

**Discussion by katelynsills** [Agoric/agoric-sdk#11225] [General]
This issue is just to keep track of the types of Zoe contracts that we know we would like to write. Each of these should be pulled off into new issues to give more specification and detail. 

* DAOs
* Negotiation
* Using derivatives as assets extensively
* Stablecoins
* Loans
* Arbitration 
* Pixel Gallery Rewrite
* Assurance Contract
* Dominant Assurance Contract

**Comment by DavidBruant** [comment]
My understanding is that the Pixel Gallery is still based on contractHost, so it might make sense to add it to the list

**Comment by katelynsills** [comment]
Yup, added.

**Comment by zarutian** [comment]
A few that come to mind but might be dependent on oracles:

- Conditional Cheque (better known as Letter of credit for some reason)
- Insurance bond
- Various kinds of auctions
- Payment cascade. (First tier gets paid first, if any left then second tier and so on)
- „King/Queen of the coin hill“ (seem quite popular for some reason)
- Grant matcher („I will donate 20 buckazoids to Amnesty Interstellar but only if you donate equally to them too“)
- n-of-k wallet/authorization/escrow
- Art-Asset Assurance contract (example: an author of an ongoing serialized story has already written a chapter but wants 200 zorkmids for it to be accessable to all, yet the payers want to be
sure they get the chapter))
- Eternity Service hosting check contracts (does the service still have spefic hosted file? if so then pay the service to host it onwards. Can be periodical.)

**Comment by katelynsills** [comment]
Thanks @zarutian, these are great!

**Comment by zarutian** [comment]
I have started on [various smart contracts](https://gist.github.com/zarutian/6ac6b7576af041dd5dc5b95f1f8d1f0f), using no tooling and only consulting the documentation.

I have yet to write tests and run them. Nor have I done exhaustive code review or correctness proofs. So, consider it pre-alpha software and use at your own risk and only with testnet value or value you do not mind lost.

**Comment by katelynsills** [comment]
> I have started on [various smart contracts](https://gist.github.com/zarutian/6ac6b7576af041dd5dc5b95f1f8d1f0f), using no tooling and only consulting the documentation.
> 
> I have yet to write tests and run them. Nor have I done exhaustive code review or correctness proofs. So, consider it pre-alpha software and use at your own risk and only with testnet value or value you do not mind lost.

That's great! Once the contracts have tests and commentary, I would love to include them in a list of examples.
